File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int getMinimumDifference(TreeNode* root) {
        int answer = INT_MAX; 
        int prev = INT_MAX; 
        helper(root,answer,prev);
        return answer; 
    }
    void helper(TreeNode* root, int& answer, int& prev){
        if(!root) return; 

        helper(root->left, answer, prev);
        if(prev != INT_MAX) answer = min(answer, abs(prev - root->val)); 
        prev = root->val; 
        helper(root->right, answer, prev); 
    }
};

File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int min_ = Integer.MAX_VALUE; 
    TreeNode prev = null; 
    public int getMinimumDifference(TreeNode root) {
        if(root == null) return 0; 
        dfs(root); 
        return min_; 
    }
    public void dfs(TreeNode root){
        if(root == null) return; 
        dfs(root.left); 
        if(prev != null){
            min_ = Math.min(min_, Math.abs(root.val - prev.val)); 
        }

        prev = root; 
        dfs(root.right); 
    }
}